extensions-core/druid-kerberos/src/main/java/io/druid/security/kerberos/KerberosAuthenticator.java

```diff
@@ -334,13 +330,14 @@ public Principal getUserPrincipal()
               };
               if (newToken && !token.isExpired() && token != AuthenticationToken.ANONYMOUS) {
                 String signedToken = mySigner.sign(token.toString());
-                tokenToAuthCookie(httpResponse,
-                                 signedToken,
-                                 getCookieDomain(),
-                                 getCookiePath(),
-                                 token.getExpires(),
-                                 !token.isExpired() && token.getExpires() > 0,
-                                 isHttps
+                createAuthCookie(
```

Comment:
@b-slim the signature of `createAuthCookie()` was changed in 2.8. Would you check this change is valid?

Reply:
Thought that copying the body of the function will avoid issues if we want to support multiple Hadoop versions. Is there any reason to not have the code living in Druid?

Reply:
Hmm, do you mean this break the compatibility with Hadoop 2.7? If so, would you elaborate on how it breaks? I didn't test, but looks like it would work because `hadoop-common.jar` is in `$DRUID_HOME/extensions/druid-kerberos/` and that `hadoop-common.jar` will be used once `druid-kerberos` extension is loaded.

Reply:
having that function inline means we can compile druid with 2.7.X and 3.0.X.

Reply:
Hmm, do we need that? I think most users will use an already compiled druid rather than compiling by themselves. I think it would be fine if this works with 2.7. I'll test it.

Reply:
@jihoonson at least am sure we do and probably other folks too...we compile against internal version to make sure that at runtime we have the same version everywhere

Reply:
Ah, all right. I'll revert this change.